cmake/onnxruntime_server.cmake

```diff
@@ -1,8 +1,54 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
-
 set(SERVER_APP_NAME "onnxruntime_server")
 
+set(gRPC_BUILD_TESTS OFF CACHE INTERNAL "Don't build tests")
+set(gRPC_GFLAGS_PROVIDER "" CACHE INTERNAL "Don't use gflags")
+set(gRPC_BENCHMARK_PROVIDER "" CACHE INTERNAL "Don't use benchmark")
+set(gRPC_ZLIB_PROVIDER "package" CACHE INTERNAL "Use preinstalled zlib library")
+set(gRPC_PROTOBUF_PROVIDER "" CACHE INTERNAL "Don't use grpc protobuf, set it manually.")
+
+
+# protobuf targets have already been included as submodules - adapted from https://github.com/grpc/grpc/blob/master/cmake/protobuf.cmake
+set(_gRPC_PROTOBUF_LIBRARY_NAME "libprotobuf")
+set(_gRPC_PROTOBUF_LIBRARIES protobuf::${_gRPC_PROTOBUF_LIBRARY_NAME})
+
+set(_gRPC_PROTOBUF_PROTOC_LIBRARIES protobuf::libprotoc)
+# extract the include dir from target's properties
+
+set(_gRPC_PROTOBUF_WELLKNOWN_INCLUDE_DIR ${REPO_ROOT}/cmake/external/protobuf/src)
+set(_gRPC_PROTOBUF_PROTOC protobuf::protoc)
+set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE $<TARGET_FILE:protobuf::protoc>)
+
+set(_gRPC_PROTOBUF_INCLUDE_DIR ${PROTOBUF_INCLUDE_DIRS})
```

Comment:
I need to do it manually because the targets are already in CMAKE. 
```
CMake Error at external/grpc/third_party/protobuf/cmake/libprotobuf.cmake:115 (add_library):
  add_library cannot create target "libprotobuf" because another target with
  the same name already exists.  The existing target is a static library
  created in source directory
  "/home/colin/src/onnxruntime/cmake/external/protobuf/cmake".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  external/grpc/third_party/protobuf/cmake/CMakeLists.txt:196 (include)
```

I think I wouldn't if it was a seperate cmakelists.txt, but that doesn't seem to be the process.